pkg/printer/sonarqube.go

```diff
@@ -0,0 +1,106 @@
+package printer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/get-woke/woke/pkg/result"
+	"github.com/get-woke/woke/pkg/rule"
+)
+
+// SonarQube is a JSON printer meant for import into SonarQube
+type SonarQube struct {
+	writer  io.Writer
+	newList bool
+}
+
+type TextRange struct {
+	StartLine   int `json:"startLine"`
+	StartColumn int `json:"startColumn"`
+	EndColumn   int `json:"endColumn"`
+}
+
+type Location struct {
+	Message   string    `json:"message"`
+	FilePath  string    `json:"filePath"`
+	TextRange TextRange `json:"textRange"`
+}
+
+type Issue struct {
+	EngineID        string   `json:"engineId"`
+	RuleID          string   `json:"ruleId"`
+	PrimaryLocation Location `json:"primaryLocation"`
+	Type            string   `json:"type"`
+	Severity        string   `json:"severity"`
+}
+
+// NewSonarQube returns a new SonarQube JSON printer
+func NewSonarQube(w io.Writer) *SonarQube {
+	return &SonarQube{writer: w, newList: true}
+}
+
+func (p *SonarQube) ShouldSkipExitMessage() bool {
+	return true
+}
+
+func calculateSonarSeverity(s rule.Severity) string {
+	// Translate the severity to sonarqube terms
+	var sonarSeverity = `MAJOR`
+	if s == rule.SevWarn {
+		sonarSeverity = `MINOR`
+	} else if s == rule.SevInfo {
+		sonarSeverity = `INFO`
+	}
+	return sonarSeverity
```

Comment:
Fixed - good call. I went with the second form because I believe the first would decrease the code coverage slightly (since the final return wouldn't be called in normal operation in that case.)